arguments/grafana.go

```diff
@@ -110,11 +160,26 @@ func grafanaDirectory(ctx context.Context, opts *pipeline.ArgumentOpts) (any, er
 		return nil, err
 	}
 
-	src, err := cloneOrMount(ctx, opts.Client, o.GrafanaDir, o.GrafanaRepo, o.GrafanaRef, o)
+	ght, err := o.githubToken(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	src, err := cloneOrMount(ctx, opts.Client, o.GrafanaDir, o.GrafanaRepo, o.GrafanaRef, ght)
 	if err != nil {
 		return nil, err
 	}
 
+	if o.PatchesRepo != "" {
+		withPatches, err := applyPatches(ctx, opts.Client, src, o.PatchesRepo, o.PatchesPath, o.PatchesRef, ght)
+		if err != nil {
+			opts.Log.Debug("patch application skipped", "error", err)
```

Comment:
Do we want to just log here or early return if patch application fails?

Reply:
it can return an error if there's no folder matching the path or if there are no patch files so i figure we should just skip. Patches are tested for compatibility before they get to this point so i'm not worried about the `git am` command failing with conflicts.